### PR TITLE
Addition of Rail Fence Cipher

### PR DESCRIPTION
**Pull request by hexaorzo**
### **Describe your change:**

- [X] Add an algorithm?
Added Rail Fence Cipher

### **Checklist:**

- [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [X] This pull request is all my own work -- I have not plagiarized.
- [X] I know that pull requests will not be merged if they fail the automated tests.
- [X] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [X] All new Java files are placed inside an existing directory.
- [X] All functions and variable names follow Java naming conventions.

### CODE REVIEW

**Comment by hexaorzo** [comment]
A contribution for Hacktoberfest 2022

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

File: src/main/java/com/thealgorithms/ciphers/RailFence.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                System.out.println("ENCRYPTED TEXT IS \n" + rf.Encryption(message, depth));
+                break;
+            case 'D':
+            case 'd':
+                System.out.println("Please enter the depth number");
+                depth = input.nextInt() % 26;
+                System.out.println("DECRYPTED TEXT IS \n" + rf.Decryption(message, depth));
+                break;
+            default:
+                System.out.println("default case");
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/ciphers/RailFence.java:85]:
```suggestion
                System.out.println("Invalid Input);
```

**Review by debasishbsws — COMMENTED** [review]
Put a link for reference of this algorithm

**Comment by hexaorzo** [comment]
Alright I'll implement all the required changes soon.

**Pull request closed without merging**